pkg/runctl/run_manager.go

```diff
@@ -366,7 +366,17 @@ func (c *runManager) setupNetworkPolicyFromConfig(ctx *runContext) error {
 		Kind:  "NetworkPolicy",
 	}
 
-	configStr := c.pipelineRunsConfig.NetworkPolicy
+	configStr := c.pipelineRunsConfig.DefaultNetworkPolicy
+
+	spec := ctx.pipelineRun.GetSpec()
+	if spec.Profiles != nil && spec.Profiles.Network != "" {
+		configStr = c.pipelineRunsConfig.NetworkPolicies[spec.Profiles.Network]
+		if configStr == "" {
+			ctx.pipelineRun.UpdateResult(v1alpha1.ResultErrorConfig)
+			return fmt.Errorf("Cannot find network policy with name %q in network config map", spec.Profiles.Network)
```

Comment:
How do you distinguish this "configuration error" from other "infrastructure errors" later on?

Reply:
Via the pipeline run's `status.result` field, which is check in the controller.